app/models/test.rb

```diff
@@ -1,15 +1,23 @@
+# frozen_string_literal: true
+
 class Test < ApplicationRecord
   belongs_to :category
   has_many :questions
   belongs_to :author, class_name: 'User'
   has_many :user_tests
   has_many :users, through: :user_tests
-  class << self
-    def tests_by_category(category)
-      Test.order('tests.title asc')
-          .joins('JOIN categories ON tests.category_id = categories.id')
-          .where(categories: { title: category })
-          .pluck(:title)
-    end
+
+  scope :tests_by_category, ->(category) { joins(:category).where(categories: { title: category }) }
+  scope :tests_entry_level, -> { where(level: 0..1) }
+  scope :tests_middle_level, -> { where(level: 2..4) }
+  scope :tests_advanced_level, -> { where(level: 5..FLOAT::INFINITY) }
+
+  validates :title, presence: true
```

Comment:
`validates :title, presence: true, uniqueness: { scope: :level,
                                  message: 'test with this name and level already exists' }`